From edd7065654fbb183f9dbd1135c22263919baa38c Mon Sep 17 00:00:00 2001
From: Luiz Felipe Kina <LuizFelipe.EiskeKina@windriver.com>
Date: Tue, 23 Apr 2024 14:32:08 -0300
Subject: [PATCH] csi-powerflex: add proxy secret in templates

Add template for proxy-server-root-certificate secret needed by the
csm-authorization sidecar.

Signed-off-by: Luiz Felipe Kina <LuizFelipe.EiskeKina@windriver.com>
---
 charts/csi-vxflexos/templates/controller.yaml | 2 +-
 charts/csi-vxflexos/templates/node.yaml       | 2 +-
 charts/csi-vxflexos/values.yaml               | 4 ++++
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/charts/csi-vxflexos/templates/controller.yaml b/charts/csi-vxflexos/templates/controller.yaml
index ace6c45..e0e8baa 100644
--- a/charts/csi-vxflexos/templates/controller.yaml
+++ b/charts/csi-vxflexos/templates/controller.yaml
@@ -459,7 +459,7 @@ spec:
             secretName: karavi-authorization-config
         - name: proxy-server-root-certificate
           secret:
-            secretName: proxy-server-root-certificate
+            secretName: {{ .Release.Name }}-proxy-server-root-certificate
         {{ end }}
         {{ end }}
 {{- if ge (int .Values.certSecretCount) 1 }}
diff --git a/charts/csi-vxflexos/templates/node.yaml b/charts/csi-vxflexos/templates/node.yaml
index 6fe85b3..1cb74bb 100644
--- a/charts/csi-vxflexos/templates/node.yaml
+++ b/charts/csi-vxflexos/templates/node.yaml
@@ -411,7 +411,7 @@ spec:
             secretName: karavi-authorization-config
         - name: proxy-server-root-certificate
           secret:
-            secretName: proxy-server-root-certificate
+            secretName: {{ .Release.Name }}-proxy-server-root-certificate
 {{ end }}
 {{ end }}
 {{- if hasKey .Values "podmon" }}
diff --git a/charts/csi-vxflexos/values.yaml b/charts/csi-vxflexos/values.yaml
index d4711c3..f0bb02d 100644
--- a/charts/csi-vxflexos/values.yaml
+++ b/charts/csi-vxflexos/values.yaml
@@ -398,6 +398,10 @@ authorization:
   # Default value: None
   proxyHost:

+  # rootCertificate: certificate used to communicate with the csm-authorization server
+  # Default value: None
+  rootCertificate:
+
   # skipCertificateValidation: certificate validation of the csm-authorization server
   # Allowed Values:
   #   "true" - TLS certificate verification will be skipped
--
2.25.1
